### PR TITLE
validate studyJob when first reconcile it

### PR DESCRIPTION
**Pull request by hougangliu**
Fixes: #297

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/308)

### CODE REVIEW

**Comment by YujiOshima** [comment]
@hougangliu Thanks! 
/lgtm

File: pkg/controller/studyjob/manifest_parser.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func getWorkerManifest(c katibapi.ManagerClient, studyID string, trial *katibapi
 	}
 	var wid string
 	if dryrun {
-		wid = "validation"
+		wid = generateRandid()
```
[the remark this answers is not in the document]
Comment (hougangliu) [reply pkg/controller/studyjob/manifest_parser.go]: yes, if I don't change it, a worker template with Name "validation"  will be taken as valid by mistake.

**Comment by hougangliu** [comment]
> it would be nice if some unit tests can be added for valid/invalid studyjob specs. May be, we can take this in a separate PR

Sure, thanks

**Pull request merged**